### PR TITLE
Ocr adapter support

### PR DESCRIPTION
**Pull request by gaya3-zipstack**
## What

OCR adapter changes to support OCR tool

## Why

 To support OCR tool

## How

...

## Relevant Docs

-

## Related Issues or PRs

-

## Dependencies Versions / Env Variables

-

## Notes on Testing

...

## Screenshots

...

## Checklist

I have read and understood the [Contribution Guidelines]().

### CODE REVIEW

File: src/unstract/sdk/ocr.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+from unstract.sdk.constants import LogLevel
+from unstract.sdk.tool.base import BaseTool
+
+
+class OCR(metaclass=ABCMeta):
+    def __init__(self, tool: BaseTool):
+        self.tool = tool
+        self.ocr_adapters = adapters
+
+    # @abstractmethod
```
Comment (johnyrahul) [inline src/unstract/sdk/ocr.py:17]: Remove if not necessary

File: src/unstract/sdk/ocr.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+                    Common.METADATA
+                ][Common.ADAPTER]
+                ocr_metadata = ocr_config.get(Common.ADAPTER_METADATA)
+                ocr_adapter_class = ocr_adapter(ocr_metadata)
+
+                return ocr_adapter_class
+
+        except Exception as e:
+            self.tool.stream_log(
+                log=f"Unable to get x2text adapter {adapter_instance_id}: {e}",
```
Comment (chandrasekharan-zipstack) [inline src/unstract/sdk/ocr.py:38]:
```suggestion
                log=f"Unable to get ocr adapter {adapter_instance_id}: {e}",
```

File: src/unstract/sdk/ocr.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+class OCR(metaclass=ABCMeta):
+    def __init__(self, tool: BaseTool):
+        self.tool = tool
+        self.ocr_adapters = adapters
+
+    # @abstractmethod
+    # def convert_to_text(self, input_file: str, basic_convert: bool = False):
+    #     pass
+
+    def get_ocr(self, adapter_instance_id: str) -> OCRAdapter:
```
Comment (chandrasekharan-zipstack) [inline src/unstract/sdk/ocr.py:21]: @gaya3-zipstack this function returns `None` as well. Change the return type to `Optional[OCRAdapter]`
Was this not caught by mypy?

Comment (gaya3-zipstack) [reply src/unstract/sdk/ocr.py:21]: Yeah wondering why! Same with X2Text. also. Thanks for pointing out. Fixing both

File: src/unstract/sdk/ocr.py (hunk 1 above)
Comment (gaya3-zipstack) [reply src/unstract/sdk/ocr.py:17]: Lines 14 and 15 are required @johnyrahul

Comment (johnyrahul) [reply src/unstract/sdk/ocr.py:18]: Talking about line 18->20

File: src/unstract/sdk/ocr.py (hunk 3 above)
Comment (gaya3-zipstack) [reply src/unstract/sdk/ocr.py]: Resolved

File: src/unstract/sdk/ocr.py (hunk 1 above)
Comment (gaya3-zipstack) [reply src/unstract/sdk/ocr.py:18]: Done

**Review by chandrasekharan-zipstack — APPROVED** [review]

**Review by chandrasekharan-zipstack — APPROVED** [review]

**Pull request merged**